### PR TITLE
MINOR: Remove redundant forwarding integration tests

### PR DESCRIPTION
**Pull request by hachikuji**
We have a few integration tests for the forwarding logic which were added prior to kraft being ready for integration testing. Now that we have enabled kraft in integration tests, these tests are redundant and can be removed.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/unit/kafka/server/MetadataRequestWithForwardingTest.scala [hunk 1]
```diff
@@ -1,111 +0,0 @@
...
-
-  @BeforeEach
-  override def setUp(testInfo: TestInfo): Unit = {
-    doSetup(testInfo, createOffsetsTopic = false)
-  }
-
-  override def enableForwarding: Boolean = true
-
-  @Test
-  def testAutoTopicCreation(): Unit = {
```
Comment (hachikuji) [inline core/src/test/scala/unit/kafka/server/MetadataRequestWithForwardingTest.scala:40]: Note that all of these tests were duplicated from `MetadataRequestTest`.

**Review by jsancio — APPROVED** [review]
Thanks for making our tests finish faster. LGTM if Jenkins is green.

**Comment by hachikuji** [comment]
cc @dengziming Minor follow-up after your patch.

**Comment by dengziming** [comment]
Thanks for this nice cleanup.

**Pull request merged**